tsdb/head.go

```diff
@@ -1837,7 +2030,7 @@ func (s *memSeries) append(t int64, v float64, appendID uint64) (success, chunkC
 		s.nextAt = computeChunkEndTime(c.minTime, c.maxTime, s.nextAt)
 	}
 	if t >= s.nextAt {
-		c = s.cut(t)
+		c = s.cutNewHeadChunk(t, chunkDiskMapper)
```

Comment:
Quite hard to deduce if we can have 2x cutNewHeadChunk (c== nil and t >= s.nextAt)

Reply:
Because we cut for `t` above, the `t` here is never going to be above `s.nextAt`, because it is set based on the timestamp passed when you cut a new chunk.

PS: it has been like that since forever, I have not changed anything, just the names.